### ISSUE TITLE
Automap Line Thickness Option

### ISSUE THREAD

**Issue by OpenRift412** [fabiangreffrath/woof#1933]
In the Kex remaster, there's an option that allows you to adjust the thickness of lines displayed in the automap. I think this would be a nice feature to have for those playing at higher resolutions that want better automap visibility.

There are essentially two ways that this could be implemented:
1. Depending on the multiplier, the automap screen resolution will be lowered, thus resulting in thicker lines.
2. Apply a sort of integer scale to automap line thickness. This is how the Kex version does it. 

In both cases, the max factor in which you can increase line thickness should be dependent on the set resolution. For example, at 200p, you're stuck at 1x thickness, at 400p, you can max out at 2x thickness, etc.

**Comment by NightFright2k19** [comment]
If there should be a manual way to adjust this, the steps should have .1 intervals since you quickly end up with the lines being too thick.

**Comment by fabiangreffrath** [comment]
Proof of concept:
```patch
--- a/src/am_map.c
+++ b/src/am_map.c
@@ -1471,7 +1471,40 @@ static void AM_drawMline
     color=0;
 
   if (AM_clipMline(ml, &fl))
+  {
     AM_drawFline(&fl, color); // draws it on frame buffer using fb coords
+
+    if (abs(fl.a.y - fl.b.y) > abs(fl.a.x - fl.b.x))
+    {
+      if (fl.a.x > 0 && fl.b.x > 0)
+      {
+        fl.a.x--;
+        fl.b.x--;
+        AM_drawFline(&fl, color);
+      }
+      else if (fl.a.x < f_w - 1 && fl.b.x < f_w - 1)
+      {
+        fl.a.x++;
+        fl.b.x++;
+        AM_drawFline(&fl, color);
+      }
+    }
+    else
+    {
+      if (fl.a.y > 0 && fl.b.y > 0)
+      {
+        fl.a.y--;
+        fl.b.y--;
+        AM_drawFline(&fl, color);
+      }
+      else if (fl.a.y < f_h - 1 && fl.b.y < f_h - 1)
+      {
+        fl.a.y++;
+        fl.b.y++;
+        AM_drawFline(&fl, color);
+      }
+    }
+  }
 }
 
 //
```

**Comment by rfomin** [comment]
This is how KEX automap lines look when the thickness is set to "auto" and the resolution is 3x:

![image](https://github.com/user-attachments/assets/ca157c49-8f11-46a1-a6bf-31bedce273de)

So I guess we have to draw several lines to achieve the same effect? They also have a nice new `player_arrow`.

**Comment by OpenRift412** [comment]
> 
> So I guess we have to draw several lines to achieve the same effect? They also have a nice new `player_arrow`.

Maybe? Not sure how that would be done while preventing gaps between the "thickening lines".

~~Personally not a fan of the new `player_arrow` lol~~

**Comment by bangstk** [comment]
This is just an opinion, but I think kex's "auto" setting is too thick, targeting the approximate thickness of 320x200. it makes more complex maps hard to read. I think if it targeted the lines to be half that thickness it would be better.

**Comment by JNechaevsky** [comment]
Agreed with @bangstk. Just have a look at NERVE's Map01 or any other heaviliy detailed maps, such bold lines are doing bad job rather than good one. Probably Fabian's approach is the most optimal one, even with MAX resolution 2x lines are giving neat and clean view of the map structure.

**Comment by NightFright2k19** [comment]
You could go for 1.5x in 1080p, possibly a bit lower, and 2x for 1600p and higher. I can't see it in front of me now, though. The distinction could be made earlier.
